src/main/java/org/folio/service/pieces/ItemRecreateInventoryService.java

```diff
@@ -0,0 +1,84 @@
+package org.folio.service.pieces;
+
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS;
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS_NAME;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.models.ItemStatus;
+import org.folio.models.pieces.PieceUpdateHolder;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.service.inventory.InventoryItemManager;
+import org.folio.service.pieces.flows.DefaultPieceFlowsValidator;
+
+public class ItemRecreateInventoryService {
+
+  protected static final int ITEM_QUANTITY = 1;
+  private static final Logger logger = LogManager.getLogger(ItemRecreateInventoryService.class);
+
+  private final InventoryItemManager inventoryItemManager;
+
+  public ItemRecreateInventoryService(InventoryItemManager inventoryItemManager) {
+    this.inventoryItemManager = inventoryItemManager;
+  }
+
+  // Return an Id of the recreated Item
+  public Future<String> recreateItemInDestinationTenant(PieceUpdateHolder holder, RequestContext srcLocCtx, RequestContext dstLocCtx) {
+    // Example Case: Member Tenant 1 (University) -> Member Tenant 2 (College)
+    // Create Item in Member Tenant 2 with the same Item Id
+    // Delete Item in Member Tenant 1 by Item Id
+    var itemFuture = Promise.<String>promise();
```

Comment:
this code can be rewritten without using promises. It simplifies code. We can use promises in complex scenarios with lot of diffrent conditions and input points.
So you can save result of inventoryItemManager in some future and make compose to delete an item